### PR TITLE
rewrite event handlers to only run data migration on database created event

### PR DESCRIPTION
**Pull request by agathanatasha**
Only run database migration on database created event
Updated unit tests

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -86,94 +89,115 @@ def _render_conf_file(self) -> None:
...
+        """
+        try:
+            process = self._container.exec(
+                ["kratos", "migrate", "sql", "-e", "--config", self._config_file_path, "--yes"],
+            )
+        except ExecError as err:
+            logger.error(f"Exited with code {err.exit_code}. Stderr: {err.stderr}")
+            return False
+
+        stdout, _ = process.wait_output()
```
Comment (nsklikas) [inline src/charm.py:117]: This may throw an `ExecError`

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -49,95 +108,137 @@ def test_update_container_correct_config(
...
     }
-    mocked_sql_migration.assert_called_once()
 
     assert yaml.safe_load(harness.charm._render_conf_file()) == expected_config
 
 
-def test_update_container_correct_pebble_layer(
-    harness, mocked_kubernetes_service_patcher, mocked_sql_migration
-) -> None:
+def test_on_pebble_ready_when_missing_database_relation(harness) -> None:
```
Comment (nsklikas) [inline tests/unit/test_charm.py:120]: Isn't this the same as `test_on_pebble_ready_service_not_started_when_database_not_created`?

**Review by nsklikas — COMMENTED** [review]
I really like the tests. Left a couple of minor comments, but overall LGTM.

File: tests/unit/test_charm.py (hunk 2 above)
Comment (agathanatasha) [reply tests/unit/test_charm.py:120]: that one focuses on testing the service is not started

**Review by nsklikas — APPROVED** [review]

**Review by natalian98 — APPROVED** [review]
lgtm

**Pull request merged**